invokeai/app/invocations/baseinvocation.py

```diff
@@ -98,11 +98,13 @@ class BaseInvocationOutput(BaseModel):
 
     _output_classes: ClassVar[set[BaseInvocationOutput]] = set()
     _typeadapter: ClassVar[Optional[TypeAdapter[Any]]] = None
+    _typeadapter_needs_update: ClassVar[bool] = False
```

Comment:
What does `_needs_update` mean here? Is this a TODO / some transitional type?

Reply:
This is an internal flag that is checked during validation of invocations. It ensures the union of invocations is up-to-date and validation is correct.

Context:
Pydantic provides two ways to validate data - a model and a [type adapter](https://docs.pydantic.dev/latest/concepts/type_adapter/). In short, type adapters encapsulate validation logic in a more flexible way than a pydantic model, and it is easier to create them dynamically.

Graphs need to validate their nodes against a union of invocations. This union must be created dynamically as the app boots up and invocations are registered. It is tricky to create a dynamic pydantic model representing the union - much easier to create a dynamic type adapter, then use that for validation. 

I needed to change how the dynamic type adapter is constructed in this PR, and encountered an edge case in unit tests where the type adapter was stale. This `_needs_update` flag is set when an invocation is registered. When the type adapter is requested in `get_typeadapter()`, we check it to see if the type adapter needs to be recreated using this flag and re-create it if necessary.